compiler/src/iree/compiler/Dialect/Flow/Transforms/OutlineDispatchRegions.cpp

```diff
@@ -27,6 +30,75 @@ namespace IREE {
 namespace Flow {
 namespace {
 
+// Estimates the evaluation cost of a linalg op using a heuristic cost model.
+static int estimateLinalgOpCost(linalg::LinalgOp op) {
+  if (op.hasDynamicShape()) {
+    // Note: bounded dynamic shapes would be interesting, if the compiler used
+    // them. For now just treat dynamic shapes as arbitrarily large.
+    return INT_MAX;
+  }
+
+  int cost = 1;
+  for (auto loopRange : op.getStaticLoopRanges()) {
+    cost *= loopRange;
+  }
+  LLVM_DEBUG(llvm::dbgs() << "// " << op->getName() << " cost: " << cost
+                          << "\n");
+  return cost;
+}
+
+static std::string loopRangeToString(int64_t loopRange) {
+  return ShapedType::isDynamic(loopRange) ? "?" : llvm::itostr(loopRange);
+}
+
+// Returns a string like "512x?x128" representing a linalg op's loop ranges.
+static std::string getLinalgOpLoopRanges(linalg::LinalgOp op) {
+  auto loopRanges = op.getStaticLoopRanges();
+  std::string outputString;
+  llvm::raw_string_ostream sstream(outputString);
+  llvm::interleave(
+      loopRanges,
+      [&](int64_t loopRange) { sstream << loopRangeToString(loopRange); },
+      [&] { sstream << "x"; });
+  return outputString;
+}
+
+static std::string summarizeLinalgOp(linalg::LinalgOp op) {
+  auto opName = op->getName().getStringRef();
+  if (!opName.consume_front("linalg.")) return "";
+  std::string opSuffix = getLinalgOpLoopRanges(op);
+  // TODO(scotttodd): include element type(s) in this string
+  return opName.str() + (opSuffix.empty() ? "" : "_" + opSuffix);
+}
+
+// Summarizes the contents of a dispatch into a short string.
+// This uses heuristics to aid developer debugging.
+static std::string summarizeDispatchWorkgroupsOp(
+    DispatchWorkgroupsOp regionOp) {
+  Operation *bestOp = NULL;
+  int bestEstimatedCost = -1;
```

Comment:
`best` is ambiguous. How about `max`? Or, you can add a comment to say that.

Reply:
Just a note: you may consider adding more instructions with a limit like 5. This may help to represent a fused dispatch region like "sum_div", "square_sum", "matmul_add_mul", so on. However, the representation might not be always clear depending on the subgraph shape in a dispatch.

Reply:
I added some comments explaining the intent and mentioning other possible approaches.